src/AspNetCoreModuleV2/CommonLib/debugutil.cpp

```diff
@@ -16,11 +16,135 @@
+    RETURN_LAST_ERROR_IF(!VerQueryValue(verData.data(), L"\\", &lpBuffer, &size));
+
+    auto verInfo = reinterpret_cast<VS_FIXEDFILEINFO *>(lpBuffer);
+    if (verInfo->dwSignature == 0xfeef04bd)
+    {
+        LPVOID pvProductName = NULL;
+        unsigned int iProductNameLen = 0;
+        RETURN_LAST_ERROR_IF(!VerQueryValue(verData.data(), _T("\\StringFileInfo\\040904b0\\FileDescription"), &pvProductName, &iProductNameLen));
+
+        DebugPrintf(ASPNETCORE_DEBUG_FLAG_INFO, "Initializing logs for %S. File Version: %d.%d.%d.%d. Description: %S",
+            path,
+            ( verInfo->dwFileVersionMS >> 16 ) & 0xffff,
+            ( verInfo->dwFileVersionMS >>  0 ) & 0xffff,
+            ( verInfo->dwFileVersionLS >> 16 ) & 0xffff,
+            ( verInfo->dwFileVersionLS >>  0 ) & 0xffff,
+            pvProductName
+        );
+    }
+
+    return S_OK;
+}
+
+void SetDebugFlags(const std::wstring &debugValue)
+{
+    try
+    {
+        if (!debugValue.empty())
+        {
+            const auto value = std::stoi(debugValue);
+
+            if (value >= 1) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_ERROR;
+            if (value >= 2) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_WARNING;
+            if (value >= 3) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_INFO;
+            if (value >= 4) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_CONSOLE;
+            if (value >= 5) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_FILE;
+
+            return;
+        }
+    }
+    catch (...)
+    {
+        // ignore
+    }
+
+    try
+    {
+        std::wstringstream stringStream(debugValue);
+        std::wstring flag;
+
+        while (std::getline(stringStream, flag, L','))
+        {
+            if (flag == L"error") DEBUG_FLAGS_VAR |= DEBUG_FLAGS_ERROR;
+            if (flag == L"warning") DEBUG_FLAGS_VAR |= DEBUG_FLAGS_WARN;
+            if (flag == L"info") DEBUG_FLAGS_VAR |= DEBUG_FLAGS_INFO;
+            if (flag == L"console") DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_CONSOLE;
+            if (flag == L"file") DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_FILE;
+        }
+
+        // If file or console is enabled but level is not set, enable all levels
+        if (DEBUG_FLAGS_VAR != 0 && (DEBUG_FLAGS_VAR & DEBUG_FLAGS_ANY) == 0)
+        {
+            DEBUG_FLAGS_VAR |= DEBUG_FLAGS_ANY;
+        }
+    }
+    catch (...)
+    {
+        // ignore
+    }
+}
+
+bool CreateDebugLogFile(const std::wstring &debugOutputFile)
+{
+    try
+    {
+        if (!debugOutputFile.empty())
+        {
+            if (g_logFile != INVALID_HANDLE_VALUE)
+            {
+                LOG_INFOF("Switching debug log files to %S", debugOutputFile.c_str());
+            }
+
+            SRWExclusiveLock lock(g_logFileLock);
+            if (g_logFile != INVALID_HANDLE_VALUE)
+            {
+                CloseHandle(g_logFile);
+                g_logFile = INVALID_HANDLE_VALUE;
+            }
+            g_logFile = CreateFileW(debugOutputFile.c_str(),
```

Comment:
Can we add the timestamp to the debug log to stay consistent with normal logs? Create a helper function that creates the file for both normal and debug?

Reply:
I don't like timestamp in file names. You have to jump through all the hoops to find the file afterwards. If we decide to do it I'll do it in separate PR because of additional deployer changes.

Reply:
I'd be okay with removing timestamps from file names.